### ISSUE TITLE
Functional Hedera network configured in HardhatUserConfig.networks

### ISSUE THREAD

**Issue by mshakeg** [hashgraph/hedera-hardhat-hethers#32]
### Problem

A hedera network configured in HardhatUserConfig.networks isn't functional presumably due to the lack of parity in the `hedera-json-rpc-relay` with the standard ethereum json rpc.

```ts
import { HardhatUserConfig } from "hardhat/config";
import "@nomicfoundation/hardhat-toolbox";

const config: HardhatUserConfig = {
  ...,
  defaultNetwork: "hedera",
  networks: {
    hedera: {
      url: "http://127.0.0.1:7546",
      chainId: 298,
      accounts: [
        // Alias ECDSA keys private keys
        "0x105d050185ccb907fba04dd92d8de9e32c18305e097ab41dadda21489a211524",
        "0x2e1d968b041d84dd120a5860cee60cd83f9374ef527ca86996317ada3d0d03e7",
        "0x45a5a7108a18dd5013cf2d5857a28144beadc9c70b3bdbd914e38df4e804b8d8",
        "0x6e9d61a325be3f6675cf8b7676c70e4a004d2308e3e182370a41f5653d52c6bd",
        "0x0b58b1bd44469ac9f813b5aeaf6213ddaea26720f0b2f133d08b6f234130a64f",
        "0x95eac372e0f0df3b43740fa780e62458b2d2cc32d6a440877f1cc2a9ad0c35cc",
        "0x6c6e6727b40c8d4b616ab0d26af357af09337299f09c66704146e14236972106",
        "0x5072e7aa1b03f531b4731a32a021f6a5d20d5ddc4e55acbb71ae202fc6f3a26d",
        "0x60fe891f13824a2c1da20fb6a14e28fa353421191069ba6b6d09dd6c29b90eff",
        "0xeae4e00ece872dd14fb6dc7a04f390563c7d69d16326f2a703ec8e0934060cc7",
      ],
    },
  }
  ...,
};

export default config;
```

Attempting to run a simple test fails on the first call to the json rpc relay i.e. `deploy`:
```ts
import { expect } from "chai";
import { ethers } from 'hardhat';

describe("Greeter", function () {
  it("Should return the new greeting once it's changed", async function () {

    const signers = await ethers.getSigners();

    const signer = signers[0];

    const Greeter = await ethers.getContractFactory("Greeter", signer);

    // fails here
    const greeter = await Greeter.deploy("Hello, world!");

    await greeter.deployed();

    expect(await greeter.greet()).to.equal("Hello, world!");

    const setGreetingTx = await greeter.setGreeting("Hola, mundo!");

    // wait until the transaction is mined
    await setGreetingTx.wait();

    expect(await greeter.greet()).to.equal("Hola, mundo!");
  });
});
```
Hardhat error stack trace:
```
ProviderError: Internal error
  at HttpProvider.request (node_modules/hardhat/src/internal/core/providers/http.ts:78:19)
  at LocalAccountsProvider.request (node_modules/hardhat/src/internal/core/providers/accounts.ts:188:34)
  at async EthersProviderWrapper.send (node_modules/@nomiclabs/hardhat-ethers/src/internal/ethers-provider-wrapper.ts:13:20)
```

Greeter.sol:
```solidity
//SPDX-License-Identifier: Unlicense
pragma solidity ^0.8.9;

contract Greeter {
    string private greeting;

    constructor(string memory _greeting) {
        greeting = _greeting;
    }

    function greet() public view returns (string memory) {
        return greeting;
    }

    function setGreeting(string memory _greeting) public {
        greeting = _greeting;
    }
}
```

json-rpc-relay docker container logs:
```
[2022-08-29 08:42:13.714 +0000] INFO (rpc-server/50 on baafe97141e2): [GET]: /health/liveness 1 ms
[2022-08-29 08:42:13.958 +0000] DEBUG (rpc-server/50 on baafe97141e2): eth_chainId
[2022-08-29 08:42:13.958 +0000] TRACE (relay-eth/50 on baafe97141e2): chainId()
[2022-08-29 08:42:13.960 +0000] INFO (rpc-server/50 on baafe97141e2): [POST] eth_chainId: 200 1 ms 
[2022-08-29 08:42:13.964 +0000] DEBUG (rpc-server/50 on baafe97141e2): eth_chainId
[2022-08-29 08:42:13.964 +0000] TRACE (relay-eth/50 on baafe97141e2): chainId()
[2022-08-29 08:42:13.964 +0000] INFO (rpc-server/50 on baafe97141e2): [POST] eth_chainId: 200 0 ms 
[2022-08-29 08:42:13.971 +0000] DEBUG (rpc-server/50 on baafe97141e2): eth_chainId
[2022-08-29 08:42:13.971 +0000] TRACE (relay-eth/50 on baafe97141e2): chainId()
[2022-08-29 08:42:13.971 +0000] INFO (rpc-server/50 on baafe97141e2): [POST] eth_chainId: 200 0 ms 
[2022-08-29 08:42:13.997 +0000] DEBUG (rpc-server/50 on baafe97141e2): eth_blockNumber
[2022-08-29 08:42:13.998 +0000] TRACE (relay-eth/50 on baafe97141e2): blockNumber()
[2022-08-29 08:42:14.024 +0000] DEBUG (rpc-server/50 on baafe97141e2): eth_chainId
[2022-08-29 08:42:14.024 +0000] TRACE (relay-eth/50 on baafe97141e2): chainId()
[2022-08-29 08:42:14.024 +0000] INFO (rpc-server/50 on baafe97141e2): [POST] eth_chainId: 200 0 ms 
[2022-08-29 08:42:14.037 +0000] ERROR (mirror-node/50 on baafe97141e2): [GET] blocks?limit=1&order=desc 567 status
    err: {
      "type": "Error",
      "message": "connect ECONNREFUSED 127.0.0.1:5551",
      "stack":
          Error: connect ECONNREFUSED 127.0.0.1:5551
              at MirrorNodeClient.handleError (/home/node/app/packages/relay/dist/lib/clients/mirrorNodeClient.js:107:27)
              at MirrorNodeClient.<anonymous> (/home/node/app/packages/relay/dist/lib/clients/mirrorNodeClient.js:95:22)
              at Generator.throw (<anonymous>)
              at rejected (/home/node/app/packages/relay/dist/lib/clients/mirrorNodeClient.js:25:65)
              at processTicksAndRejections (node:internal/process/task_queues:96:5)
    }
[2022-08-29 08:42:14.042 +0000] ERROR (Internal error/50 on baafe97141e2): [POST] eth_blockNumber: -32603 45 ms
    code: -32603
    message: "Unknown error invoking RPC"
```

### Solution

- Requires further investigation.

**Comment by dimitrovmaksim** [comment]
I think you have to setup the hedera network in the hedera network config
```
const config: HardhatUserConfig = {
  ...,
  defaultNetwork: "hedera",
  hedera: {
     networks: {
        hedera: { ... }
    }
}
```

and not in the hardhat network config as stated in the config https://github.com/hashgraph/hedera-hardhat-hethers#configuration

**Comment by mshakeg** [comment]
I'm aware of that, however that requires using the `@hashgraph/hardhat-hethers` plugin. However, I would like to connect to the local hedera json rpc directly as one would for any other local ethereum node e.g. ganache, etc.

The reason is that using the `@hashgraph/hardhat-hethers` plugin seems to break other hardhat plugins as I've pointed out in some of the other issues I created: #28 #25 

In addition `hethers` doesn't have all the functionality that `hardhat.ethers` has, as pointed out in these issues: #30 #29 #27
